server/image.go

```diff
@@ -189,9 +198,55 @@ func (im *Image) Release() {
 	}
 }
 
-func (im *Image) sync() {
+func (im *Image) sync() error {
 	im.Width = im.img.Width()
 	im.Height = im.img.Height()
+
+	if im.Format == "mp4" && len(im.Data) != 0 {
+		// No need to convert again.
+		return nil
+	}
+
 	im.Format = im.img.Format()
 	im.Data = im.img.Data()
+
+	if im.conversionFormat == "mp4" && im.Format == "gif" {
```

Comment:
btw, here is a function where we call out to exiftool and pass data to it via stdin -- https://gist.github.com/pkieltyka/5f565ee8510c4091f381

perhaps this will help you figure out how to pipe gif data to ffmpeg